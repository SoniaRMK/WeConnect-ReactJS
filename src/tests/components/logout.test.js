import {shallow, mount} from 'enzyme';
import React from 'react';
import Weconnect from '../../store';
import { MemoryRouter } from 'react-router-dom';
import mockSessionStorage from '../sessionStorage';
import decode from 'jwt-decode';
import jwt from "jsonwebtoken";
import LogoutUser from '../../components/user/logout';


const loginUserMock = { token: jwt.sign({ user_email: "sonia@yahoo.com", user_password: "1234567890" }, "Oxa34KLncvfjKEjXkf") 
                   };
describe('user reset password component',() => {

    beforeEach(() => {
        Object.defineProperty(window, "sessionStorage", {
            value: mockSessionStorage
        });
        sessionStorage.setItem("access_token", loginUserMock.token);
    });

    it('should match snapshot',() =>{
        const wrapper = shallow(<LogoutUser/>)
        expect(wrapper).toMatchSnapshot();
    })

    it("should render the logout component", () => {
        const userComponent = mount(<MemoryRouter><LogoutUser store = {Weconnect}/></MemoryRouter>)
        expect(userComponent.length).toBe(1);
        
    });
})
