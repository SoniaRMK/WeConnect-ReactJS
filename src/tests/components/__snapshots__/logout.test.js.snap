// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user reset password component should match snapshot 1`] = `
<Route
  render={[Function]}
/>
`;
